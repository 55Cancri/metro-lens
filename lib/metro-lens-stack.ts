import * as sns from "@aws-cdk/aws-sns"
import * as subs from "@aws-cdk/aws-sns-subscriptions"
import * as sqs from "@aws-cdk/aws-sqs"
import * as cdk from "@aws-cdk/core"
import { Bucket, BlockPublicAccess } from "@aws-cdk/aws-s3"
import { BucketDeployment, Source } from "@aws-cdk/aws-s3-deployment"
import { CloudFrontWebDistribution } from "@aws-cdk/aws-cloudfront"
import { RemovalPolicy } from "@aws-cdk/core"

export class MetroLensStack extends cdk.Stack {
    constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
        super(scope, id, props)

        // source the spa files
        const source = Source.asset("../client/build")

        // create the s3 bucket
        const bucket = this.createBucket()

        // put spa in s3 bucket
        this.deploySourceToBucket(bucket)

        // create cloudfront
        this.createCloudFront(bucket)

        // create route 53

        // potentially apigateway

        // potentially lambda to handle requests

        // potentially dynamodb if users or metrics should be recorded

        // const queue = new sqs.Queue(this, 'MetroLensQueue', {
        //   visibilityTimeout: cdk.Duration.seconds(300)
        // })

        // const topic = new sns.Topic(this, 'MetroLensTopic')

        // topic.addSubscription(new subs.SqsSubscription(queue))
    }

    private createBucket = () =>
        new Bucket(this, "client", {
            bucketName: "metro-lens-client",
            websiteIndexDocument: "index.html",
            websiteErrorDocument: "index.html",
            blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
            // completely destroy bucket with cdk destroy
            removalPolicy: RemovalPolicy.DESTROY,
            // only cloudfront can read the bucket
            publicReadAccess: false,
            // upload a new file on each upload
            versioned: true,
        })

    private deploySourceToBucket = (bucket: Bucket, source: Source) =>
        new BucketDeployment(this, `BucketDeployment`, {
            sources: [source],
            destinationBucket: bucket,
        })

    private createCloudFront = (bucket: Bucket, hostedZone) => {
        // // ?
        // const domainName = `${subdomain}.${hostedZone.zoneName}`

        // // ?
        // const certificate = new DnsValidatedCertificate(this, 'SiteCertificate', {
        //   region: 'us-east-1',
        //   domainName,
        //   hostedZone
        // })

        // // ?
        // new CfnOutput(this, 'URL', { value: `https://${domainName}/` })

        return new CloudFrontWebDistribution(this, "CloudFront", {
            // ?
            // viewerCertificate: ViewerCertificate.fromAcmCertificate(certificate, {
            //   // ?
            //   aliases: [domainName]
            // }),
            originConfigs: [
                {
                    s3OriginSource: { s3BucketSource: bucket },
                    behaviors: [{ isDefaultBehavior: true }],
                },
            ],
        })
    }
}
