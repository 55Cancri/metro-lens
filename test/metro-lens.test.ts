import { expect as expectCDK, haveResource, SynthUtils } from "@aws-cdk/assert"
import * as cdk from "@aws-cdk/core"
import * as path from "path"
import * as dotenv from "dotenv"
import "jest-cdk-snapshot"

import { MetroLensStack } from "../lib/metro-lens-stack"

/* define the path to the environment file */
const toConfig = path.resolve(__dirname, "../staging") + "/.env.atlantic"

/* read the file in */
dotenv.config({ path: toConfig })

const appName = "metro-lens"
const UI_DIRECTORY = "client/build"

/* define the stack props */
const props = {
  appName,
  uiDirectory: UI_DIRECTORY,
  environmentName: process.env.ENV_NAME!,
  resourcePrefix: `${process.env.ENV_NAME!}-${appName}`,
  hostedZoneId: process.env.HOSTED_ZONE_ID!,
  hostedZoneName: process.env.HOSTED_ZONE_NAME!,
  aliasRecordName: process.env.UI_DOMAIN_ALIAS!,
  env: {
    account: process.env.CDK_DEFAULT_ACCOUNT,
    region: process.env.CDK_DEFAULT_REGION,
  },
}

test("S3 Bucket Created", () => {
  const app = new cdk.App()
  // WHEN
  const stack = new MetroLensStack(app, "MyTestStack", props)
  // THEN
  expect(stack).toMatchCdkSnapshot()
})

test("CloudFront Created", () => {
  const app = new cdk.App()
  // WHEN
  const stack = new MetroLensStack(app, "MyTestStack", props)
  // THEN
  expect(stack).toMatchCdkSnapshot()
})

// test('SQS Queue Created', () => {
//     const app = new cdk.App();
//     // WHEN
//     const stack = new MetroLens.MetroLensStack(app, 'MyTestStack');
//     // THEN
//     expectCDK(stack).to(haveResource("AWS::SQS::Queue",{
//       VisibilityTimeout: 300
//     }));
// });

// test('SNS Topic Created', () => {
//   const app = new cdk.App();
//   // WHEN
//   const stack = new MetroLens.MetroLensStack(app, 'MyTestStack');
//   // THEN
//   expectCDK(stack).to(haveResource("AWS::SNS::Topic"));
// });
