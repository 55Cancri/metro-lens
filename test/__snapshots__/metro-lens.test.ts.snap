// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloudFront Created 1`] = `
Object {
  "Parameters": Object {
    "AssetParametersa9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ffArtifactHash08605F5E": Object {
      "Description": "Artifact hash for asset \\"a9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ff\\"",
      "Type": "String",
    },
    "AssetParametersa9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ffS3Bucket848A1F31": Object {
      "Description": "S3 bucket for asset \\"a9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ff\\"",
      "Type": "String",
    },
    "AssetParametersa9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ffS3VersionKey983DBE96": Object {
      "Description": "S3 key for asset version \\"a9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ff\\"",
      "Type": "String",
    },
    "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dArtifactHash97D56BBE": Object {
      "Description": "Artifact hash for asset \\"c5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1d\\"",
      "Type": "String",
    },
    "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3Bucket19D75DB8": Object {
      "Description": "S3 bucket for asset \\"c5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1d\\"",
      "Type": "String",
    },
    "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3VersionKeyA777FDD1": Object {
      "Description": "S3 key for asset version \\"c5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1d\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "AliasRecord851000D2": Object {
      "Properties": Object {
        "AliasTarget": Object {
          "DNSName": Object {
            "Fn::GetAtt": Array [
              "CloudFrontCFDistribution57EFBAC6",
              "DomainName",
            ],
          },
          "HostedZoneId": "Z2FDTNDATAQYW2",
        },
        "HostedZoneId": "DUMMY",
        "Name": "metro-lens.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "BucketDeploymentCustomResourceCB7E2133": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DestinationBucketName": Object {
          "Ref": "atlanticmetrolensclientA0FCA384",
        },
        "DistributionId": Object {
          "Ref": "CloudFrontCFDistribution57EFBAC6",
        },
        "DistributionPaths": Array [
          "/*",
        ],
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": Array [
          Object {
            "Ref": "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3Bucket19D75DB8",
          },
        ],
        "SourceObjectKeys": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3VersionKeyA777FDD1",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3VersionKeyA777FDD1",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
    "CloudFrontCFDistribution57EFBAC6": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "Aliases": Array [
            "www.undefined",
          ],
          "Comment": "The cloudfront distribution for metro-lens.com.",
          "CustomErrorResponses": Array [
            Object {
              "ErrorCachingMinTTL": 300,
              "ErrorCode": 403,
              "ResponseCode": 200,
              "ResponsePagePath": "/index.html",
            },
            Object {
              "ErrorCachingMinTTL": 300,
              "ErrorCode": 404,
              "ResponseCode": 200,
              "ResponsePagePath": "/index.html",
            },
          ],
          "DefaultCacheBehavior": Object {
            "AllowedMethods": Array [
              "GET",
              "HEAD",
            ],
            "CachedMethods": Array [
              "GET",
              "HEAD",
            ],
            "Compress": true,
            "ForwardedValues": Object {
              "Cookies": Object {
                "Forward": "none",
              },
              "QueryString": false,
            },
            "TargetOriginId": "origin1",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "DomainName": Object {
                "Fn::GetAtt": Array [
                  "atlanticmetrolensclientA0FCA384",
                  "RegionalDomainName",
                ],
              },
              "Id": "origin1",
              "S3OriginConfig": Object {
                "OriginAccessIdentity": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "origin-access-identity/cloudfront/",
                      Object {
                        "Ref": "OAI",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "PriceClass": "PriceClass_100",
          "ViewerCertificate": Object {
            "AcmCertificateArn": "arn:aws:acm:us-east-1:707446208631:certificate/6311befd-39f0-48a6-be8d-1f8274f77371",
            "MinimumProtocolVersion": "TLSv1.2_2018",
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": Object {
      "DependsOn": Array [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersa9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ffS3Bucket848A1F31",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersa9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ffS3VersionKey983DBE96",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersa9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ffS3VersionKey983DBE96",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.6",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3Bucket19D75DB8",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3Bucket19D75DB8",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "atlanticmetrolensclientA0FCA384",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "atlanticmetrolensclientA0FCA384",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "cloudfront:GetInvalidation",
                "cloudfront:CreateInvalidation",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": Array [
          Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "OAI": Object {
      "Properties": Object {
        "CloudFrontOriginAccessIdentityConfig": Object {
          "Comment": "Necessary for CloudFront to gain access to the bucket.",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "atlanticmetrolensclientA0FCA384": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketName": "atlantic-metro-lens-client",
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "VersioningConfiguration": Object {
          "Status": "Enabled",
        },
        "WebsiteConfiguration": Object {
          "ErrorDocument": "index.html",
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "atlanticmetrolensclientPolicy2B31514D": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "atlanticmetrolensclientA0FCA384",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::cloudfront:user/CloudFront Origin Access Identity ",
                      Object {
                        "Ref": "OAI",
                      },
                    ],
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "atlanticmetrolensclientA0FCA384",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "atlanticmetrolensclientA0FCA384",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
}
`;

exports[`S3 Bucket Created 1`] = `
Object {
  "Parameters": Object {
    "AssetParametersa9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ffArtifactHash08605F5E": Object {
      "Description": "Artifact hash for asset \\"a9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ff\\"",
      "Type": "String",
    },
    "AssetParametersa9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ffS3Bucket848A1F31": Object {
      "Description": "S3 bucket for asset \\"a9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ff\\"",
      "Type": "String",
    },
    "AssetParametersa9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ffS3VersionKey983DBE96": Object {
      "Description": "S3 key for asset version \\"a9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ff\\"",
      "Type": "String",
    },
    "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dArtifactHash97D56BBE": Object {
      "Description": "Artifact hash for asset \\"c5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1d\\"",
      "Type": "String",
    },
    "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3Bucket19D75DB8": Object {
      "Description": "S3 bucket for asset \\"c5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1d\\"",
      "Type": "String",
    },
    "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3VersionKeyA777FDD1": Object {
      "Description": "S3 key for asset version \\"c5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1d\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "AliasRecord851000D2": Object {
      "Properties": Object {
        "AliasTarget": Object {
          "DNSName": Object {
            "Fn::GetAtt": Array [
              "CloudFrontCFDistribution57EFBAC6",
              "DomainName",
            ],
          },
          "HostedZoneId": "Z2FDTNDATAQYW2",
        },
        "HostedZoneId": "DUMMY",
        "Name": "metro-lens.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "BucketDeploymentCustomResourceCB7E2133": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DestinationBucketName": Object {
          "Ref": "atlanticmetrolensclientA0FCA384",
        },
        "DistributionId": Object {
          "Ref": "CloudFrontCFDistribution57EFBAC6",
        },
        "DistributionPaths": Array [
          "/*",
        ],
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": Array [
          Object {
            "Ref": "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3Bucket19D75DB8",
          },
        ],
        "SourceObjectKeys": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3VersionKeyA777FDD1",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3VersionKeyA777FDD1",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
    "CloudFrontCFDistribution57EFBAC6": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "Aliases": Array [
            "www.undefined",
          ],
          "Comment": "The cloudfront distribution for metro-lens.com.",
          "CustomErrorResponses": Array [
            Object {
              "ErrorCachingMinTTL": 300,
              "ErrorCode": 403,
              "ResponseCode": 200,
              "ResponsePagePath": "/index.html",
            },
            Object {
              "ErrorCachingMinTTL": 300,
              "ErrorCode": 404,
              "ResponseCode": 200,
              "ResponsePagePath": "/index.html",
            },
          ],
          "DefaultCacheBehavior": Object {
            "AllowedMethods": Array [
              "GET",
              "HEAD",
            ],
            "CachedMethods": Array [
              "GET",
              "HEAD",
            ],
            "Compress": true,
            "ForwardedValues": Object {
              "Cookies": Object {
                "Forward": "none",
              },
              "QueryString": false,
            },
            "TargetOriginId": "origin1",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "DomainName": Object {
                "Fn::GetAtt": Array [
                  "atlanticmetrolensclientA0FCA384",
                  "RegionalDomainName",
                ],
              },
              "Id": "origin1",
              "S3OriginConfig": Object {
                "OriginAccessIdentity": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "origin-access-identity/cloudfront/",
                      Object {
                        "Ref": "OAI",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "PriceClass": "PriceClass_100",
          "ViewerCertificate": Object {
            "AcmCertificateArn": "arn:aws:acm:us-east-1:707446208631:certificate/6311befd-39f0-48a6-be8d-1f8274f77371",
            "MinimumProtocolVersion": "TLSv1.2_2018",
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": Object {
      "DependsOn": Array [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersa9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ffS3Bucket848A1F31",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersa9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ffS3VersionKey983DBE96",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersa9125fa9a40550c71cde90bd478cc23091e868067a12380c1df0827d013ad2ffS3VersionKey983DBE96",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.6",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3Bucket19D75DB8",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "AssetParametersc5fef33afe08c77f89dde679c697be27c4e2b108d12823c8331494893fad8f1dS3Bucket19D75DB8",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "atlanticmetrolensclientA0FCA384",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "atlanticmetrolensclientA0FCA384",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "cloudfront:GetInvalidation",
                "cloudfront:CreateInvalidation",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": Array [
          Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "OAI": Object {
      "Properties": Object {
        "CloudFrontOriginAccessIdentityConfig": Object {
          "Comment": "Necessary for CloudFront to gain access to the bucket.",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "atlanticmetrolensclientA0FCA384": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketName": "atlantic-metro-lens-client",
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "VersioningConfiguration": Object {
          "Status": "Enabled",
        },
        "WebsiteConfiguration": Object {
          "ErrorDocument": "index.html",
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "atlanticmetrolensclientPolicy2B31514D": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "atlanticmetrolensclientA0FCA384",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::cloudfront:user/CloudFront Origin Access Identity ",
                      Object {
                        "Ref": "OAI",
                      },
                    ],
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "atlanticmetrolensclientA0FCA384",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "atlanticmetrolensclientA0FCA384",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
}
`;
